### PR TITLE
Update .travis.yml

### PR DESCRIPTION
**Pull request by odow**
Removes an unused key from Travis, with the side-effect of seeing if the documentation will build. 

x-ref https://github.com/jump-dev/jump-dev.github.io/pull/30#issuecomment-692346731

### CODE REVIEW

**Comment by mlubin** [comment]
Here's how our config changed since the last release: https://github.com/jump-dev/JuMP.jl/commit/29f64d81071ad2169502c704e53ae1cfaaec0f3f. Maybe the "branches" section had some effect?

**Comment by odow** [comment]
@mlubin I added depwarn. Let's see if it works.

**Comment by mlubin** [comment]
Well, ``--depwarn=error` is going to fail before https://github.com/jump-dev/JuMP.jl/pull/2329 is merged.

**Comment by odow** [comment]
> Well, `--depwarn=error` is going to fail before #2329 is merged.

Yes. If we added it to your PR, it would pass and we wouldn't know if it was the fix or whether the argument was ignored.

**Comment by odow** [comment]
On a different note, I seem to have lost access to the JuMP-dev org on travis. I can't restart builds. On `travis-ci.org`, it says I don't belong to organizations, but on `travis-ci.com` it says I belong to JuMP-dev. But JuMP.jl is on `travis-ci.org`. Did you have any issues?

**Comment by mlubin** [comment]
Oh, we also moved orgs since the last release. Maybe that caused the failure to push new docs.

No, I haven't had permissions issues with travis-ci.org. You are an owner of the JuMP-dev org so not sure what else to do.

**Comment by odow** [comment]
Ah, I didn't read the error message closely enough. Although `MbedTLS` usually fails due to network issues, this time it is because of a deprecation warning:
https://travis-ci.org/github/jump-dev/JuMP.jl/jobs/727217390#L449
I guess `depwarn=error` isn't a good idea then.

**Comment by mlubin** [comment]
`@fredrikekre` said on slack:
> You are not building on tags
> This filters them out: https://github.com/jump-dev/JuMP.jl/blob/a4592816023982d9979ebcd16bfc9e3bbac1f38e/.travis.yml#L18-L21
> That branch filter was not there on 0.21.3
> You need to add something like https://github.com/fredrikekre/Literate.jl/blob/5db8c22fcfe74470d1ab836d566afbc8a9e86d9f/.travis.yml#L13 to also run travis on the tags

I think we can manually push the docs for 0.21.4 if we want. But we could also fix this and then tag 0.21.5 to get https://github.com/jump-dev/JuMP.jl/pull/2329.

**Comment by odow** [comment]
Agree on a `0.21.5`. We also need a `0.9.16` for MOI.

File: .travis.yml [hunk 1]
```diff
@@ -19,14 +16,19 @@ branches:
   only:
     - master
     - /^release-.*$/
+    # Regex to allow building Documentation
```
Comment (mlubin) [inline .travis.yml:19]: I'd word this as: "Matches tags (needed to push documentation)"

**Comment by odow** [comment]
TODO: when merging, remove `[ci skip]` from commit message to ensure `master` build is triggered.

File: .travis.yml [hunk 2]
```diff
@@ -19,14 +16,19 @@ branches:
   only:
     - master
     - /^release-.*$/
+    # Matches tags (needed to build documentation)
+    - /^v\d+\.\d+(\.\d+)?(-\S*)?$/
 notifications:
     email: false
-sudo: false
```
Comment (mlubin) [inline .travis.yml:24]: I don't know what the effect of removing this is.

**Review by mlubin — APPROVED** [review]
LGTM assuming travis passes.

File: .travis.yml (hunk 2 above)
Comment (odow) [reply .travis.yml:24]: Travis told me it was okay
image

Comment (blegat) [reply .travis.yml:24]: sudo: true used to be the default but that changed a few years ago so it's safe to remove

**Review by blegat — APPROVED** [review]

**Pull request merged**